sample-tests/build.gradle.kts

```
@@ -7,18 +9,32 @@ repositories {
 }
 
 dependencies {
-    testImplementation("org.codehaus.groovy:groovy-all:2.5.8")
+    testImplementation(localGroovy())
     testImplementation(gradleTestKit())
-    testImplementation("org.gradle:sample-check:0.9.0")
+    testImplementation("org.gradle:sample-check:0.12.6")
 }
 
-val copySamples by tasks.registering(Sync::class) {
-    from("../samples")
-    into("$buildDir/samples")
-}
+val snippetsDir = file("../samples")
+val processedSnippetsDir = file("$buildDir/samples")
+
+val tokens: Map<String, Provider<String>> = mapOf(
+    "test-retry-plugin-version" to provider {
+        PluginsVersionData.latestVersion("org.gradle", "test-retry-gradle-plugin")
+    }
+)
+
 
-tasks.test {
-    inputs.files(copySamples)
-        .withPathSensitivity(PathSensitivity.RELATIVE)
-        .withPropertyName("samples")
+tasks {
+    val replaceTokensInSnippets by registering(Copy::class) {
+        from(snippetsDir)
+        into(processedSnippetsDir)
+        inputs.properties(tokens)
+        doFirst {
+            filter(org.apache.tools.ant.filters.ReplaceTokens::class, "tokens" to tokens.mapValues { it.value.get() })
+        }
+    }
+    test {
+        val fileProvider = project.objects.directoryProperty().fileProvider(replaceTokensInSnippets.map { it.destinationDir })
+        inputs.files(fileProvider)
```

We should preserve the `PathSensitivity` otherwise the default is `ABSOLUTE` which breaks cacheability, right?

```suggestion
        inputs.files(fileProvider)
            .withPathSensitivity(PathSensitivity.RELATIVE)
            .withPropertyName("samples")
```

---

great suggestion - fixed